aptos-move/aptos-vm/src/verifier/resource_groups.rs

```diff
@@ -145,30 +160,27 @@ pub(crate) fn validate_module_and_extract_new_entries(
 
 /// Given a module id extract all resource group metadata
 pub(crate) fn extract_resource_group_metadata_from_module(
-    module_storage: &impl AptosModuleStorage,
-    module_id: &ModuleId,
+    old_module_if_exits: Option<Arc<CompiledModule>>,
 ) -> VMResult<(
     BTreeMap<String, ResourceGroupScope>,
     BTreeMap<String, StructTag>,
     BTreeSet<String>,
 )> {
-    let module =
-        module_storage.fetch_existing_deserialized_module(module_id.address(), module_id.name());
-    let (metadata, module) = if let Ok(module) = module {
-        (get_metadata_from_compiled_code(module.as_ref()), module)
-    } else {
-        // Maintaining backwards compatibility with no validation of deserialization.
-        return Ok((BTreeMap::new(), BTreeMap::new(), BTreeSet::new()));
+    let old_module = match old_module_if_exits {
+        Some(old_module) => old_module,
+        None => {
+            return Ok((BTreeMap::new(), BTreeMap::new(), BTreeSet::new()));
```

Comment:
This makes me think the whole block belongs on the caller side.
Something like when we get maybe_old_module we actually do map_or_else or map_or_default, and in the map case we call extract_resource_group_metadata_from_module to assign to those BTreeMap/Sets.

Reply:
Good point, extracted out and changed to `map_or_else`